plenario/api/shape.py

```diff
@@ -1,76 +1,48 @@
+from plenario.utils.ogr2ogr import OgrExport
 
 
-def export_dataset_to_response(dataset_name, query=None):
-
-    """
-    :param dataset_name: Name of shape dataset. Expected to be found in meta_shape table.
-    :param query: Optional SQL query to be executed on shape dataset to filter results
-    Expected query parameter: `data_type`. We expect it to be one of 'json', 'kml', or 'shapefile'.
-                                If none of these (or unspecified), return JSON.
-    :return: response with geoJSON data and response code
-    """
-
-    # Do we have this shape?
-    shape_dataset = session.query(ShapeMetadata).get(dataset_name)
-    if not (shape_dataset and shape_dataset.is_ingested):
-        error_message = 'Could not find shape dataset {}'.format(dataset_name)
-        return make_response(error_message, 404)
+# ====================
+# Shape Format Headers
+# ====================
 
-    # What file format does the user want it in?
-    export_format = request.args.get('data_type')
-    # json is default export type
-    if not export_format:
-        export_format = u'json'
-    export_format = unicode.lower(export_format)
-
-    # Make a filename that we are reasonably sure to be unique and not occupied by anyone else.
-    sacrifice_file = tempfile.NamedTemporaryFile()
-    export_path = sacrifice_file.name
-    sacrifice_file.close()  # Removes file from system.
-
-    try:
-        # Write to that filename
-        OgrExport(export_format=export_format, table_name=dataset_name, export_path=export_path, query=query).write_file()
-        # Dump it in the response
-        with open(export_path, 'r') as to_export:
-            resp = make_response(to_export.read(), 200)
+def _shape_format_to_content_header(requested_format):
+    format_map = {
+        'json': 'application/json',
+        'kml': 'application/vnd.google-earth.kml+xml',
+        'shapefile': 'application/zip'
+    }
+    return format_map[requested_format]
 
-        # Make the downloaded filename look nice
-        resp.headers['Content-Type'] = _shape_format_to_content_header(export_format)
-        disp_header = 'attachment; filename={name}.{ext}'.format(name=shape_dataset.human_name,
-                                                                 ext=_shape_format_to_file_extension(export_format))
-        resp.headers['Content-Disposition'] = disp_header
-        return resp
-    except Exception as e:
-        error_message = 'Failed to export shape dataset {}'.format(dataset_name)
-        print repr(e)
-        return make_response(error_message, 500)
-    finally:
-        # Don't leave that file hanging around.
-        if os.path.isfile(export_path):
-            os.remove(export_path)
 
+def _shape_format_to_file_extension(requested_format):
+    format_map = {
+        'json': 'json',
+        'kml': 'kml',
+        'shapefile': 'zip'
+    }
+    return format_map[requested_format]
 
-def make_sql_ready_geom(geojson_str):
```

Comment:
this method was only ever called once in `_get_all_shape_datasets`, I figured I'd just delete it